app/models/spree/log_entry.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module Spree
+  class LogEntry < ActiveRecord::Base
+    belongs_to :source, polymorphic: true
+
+    # Fix for Spree #1767
+    # If a payment fails, we want to make sure we keep the record of it failing
+    after_rollback :save_anyway
```

Comment:
Thinking out loud: It would've been much easier not to include this in the same DB transaction so it wouldn't have been rolled back.